ingest/src/main/scala/hydra.ingest/http/TopicDeletionEndpoint.scala

```diff
@@ -0,0 +1,146 @@
+      successfulTopics.map(topicName => log.info(s"User $userDeleting deleted topic $topicName"))
+      complete(StatusCodes.Accepted, response)
+    }
+  }
+
+  private def invalidResponse(topics: List[String], userDeleting: String, e: NonEmptyList[DeleteTopicError])(implicit ec: ExecutionContext) = {
+    val allErrors = e.foldLeft((List.empty[SchemaDeletionErrors], List.empty[KafkaDeletionErrors])) { (agg, i) =>
+      i match {
+        case sde: SchemaDeletionErrors => (agg._1 :+ sde, agg._2)
+        case kde: KafkaDeletionErrors => (agg._1, agg._2 :+ kde)
+      }
+    }
+    val response = tupleErrorsToResponse(allErrors)
+    returnResponse(topics, userDeleting, response)
+  }
+
+  private def deleteTopics(topics: List[String], userDeleting: String)(implicit ec: ExecutionContext) = {
+    onComplete(
+      Futurable[F].unsafeToFuture(deletionProgram.deleteTopic(topics))
+    ) {
+      case Success(maybeSuccess) => {
+        maybeSuccess match {
+          case Validated.Valid(a) => {
+            validResponse(topics, userDeleting)
+          }
+          case Validated.Invalid(e) => {
+            invalidResponse(topics, userDeleting, e)
+          }
+        }
+      }
+      case Failure(e) => {
+        topics.map(topicName => addPromHttpMetric(topicName, StatusCodes.InternalServerError.toString(), "/deleteTopics/"))
+        complete(StatusCodes.InternalServerError, e.getMessage)
+      }
+    }
+  }
+
+  def myUserPassAuthenticator(credentials: Credentials): Option[String] =
+    credentials match {
+      case p@Credentials.Provided(id) if p.verify(deletionPassword) => Some(id)
+      case _ => None
+    }
+
+  override val route: Route =
+    handleExceptions(exceptionHandler) {
+      extractExecutionContext { implicit ec =>
+        pathPrefix("v2" / "topics") {
+          delete {
+            authenticateBasic(realm = "secure site", myUserPassAuthenticator) { userName =>
+              pathPrefix("schemas" / Segment) { topic =>
+                onComplete(
+                  Futurable[F].unsafeToFuture(deletionProgram.deleteFromSchemaRegistry(List(topic)))
+                ) {
+                  case Success(maybeSuccess) => {
+                    maybeSuccess match {
+                      case Validated.Valid(a) => {
+                        validResponse(List(topic), userName)
+                      }
+                      case Validated.Invalid(e) => {
+                        val response = schemaErrorsToResponse(List(SchemaDeletionErrors(SchemaDeleteTopicErrorList(e))))
+                        returnResponse(List(topic), userName, response)
+                      }
+                    }
+                  }
+                }
+              } ~
+              pathPrefix(Segment) { topic =>
+                deleteTopics(List(topic), userName)
+              } ~
+              pathEndOrSingleSlash {
+                entity(as[DeletionRequest]) { req =>
+                  val maybeList = req.topics
```

Comment:
Nit: this val is probably not helpful as it doesn't really describe the topics better. I would probably just pass `req.topics` direct to `deleteTopics`